src/main/scala/TextToStructuredData.scala

```diff
@@ -0,0 +1,25 @@
+import org.apache.spark.sql.functions._
+import TweetsNormalization.normalizeTweets
+import FeaturesExtractionFromRawTweet.extractFeaturesFromRawTweet
+import org.apache.spark.sql.{DataFrame, Dataset}
+
+object TextToStructuredData extends SparkSessionCreator {
+  def getStructuredDataFromText(text: Dataset[String]): DataFrame = {
+    import spark.implicits._
+
+    val aggregatedText = text.agg(concat_ws("\n", collect_list("value"))).as[String]
+
+    val tweets = aggregatedText
+      .flatMap(line => line.split("List\\("))
+
+    val nonEmptyTweets = tweets
+      .map(line => line.stripLineEnd)
+      .filter(line => line.length() > 0)
+
+    val nonEmptyTweetsNormalized = nonEmptyTweets.transform[String](normalizeTweets())
+
+    nonEmptyTweetsNormalized
+      .map(extractFeaturesFromRawTweet)
+      .withColumn("id", monotonically_increasing_id())
```

Comment:
could you clarify please, is there a real need to the `id`? where is it utilized?

Reply:
There are no needs for it. It's just for 'code style' :) Deleted

Reply:
So, is `StructuredTweet` sufficient? i.e. no need to use plain `Row` as a model?

Reply:
I don't understand your question. Could you please clarify what are you asking about?